src/steps/device-groups/index.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  createDirectRelationship,
+  Entity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+import {
+  ACCOUNT_ENTITY_KEY,
+  Entities,
+  Relationships,
+  Steps,
+} from '../constants';
+import { createDeviceGroupEntity } from './converter';
+
+export async function fetchDeviceGroups({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await apiClient.iterateDeviceGroups(async (deviceGroup) => {
+    const deviceGroupEntity = createDeviceGroupEntity(deviceGroup);
+
+    if (deviceGroupEntity) {
+      await jobState.addEntity(deviceGroupEntity);
+
+      await jobState.addRelationship(
+        createDirectRelationship({
+          _class: RelationshipClass.HAS,
+          from: (await jobState.getData(ACCOUNT_ENTITY_KEY)) as Entity,
+          to: deviceGroupEntity,
+        }),
+      );
+    }
+  });
+}
+
+export async function buildDeviceGroupsAndDevicesRelationships({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await jobState.iterateEntities(
+    { _type: Entities.DEVICE_GROUP._type },
+    async (deviceGroupEntity) => {
+      const deviceGroupDetails = await apiClient.fetchDeviceGroupDetails(
+        deviceGroupEntity.id as string,
+      );
+
+      if (deviceGroupDetails.devices)
+        for (const device of deviceGroupDetails.devices) {
+          const deviceEntity = await jobState.findEntity(
+            `${device.name}-${device.id}`,
```

Comment:
This generally looks great, just again the key related comment - a function that returns this would be great since it's used elsewhere than just in the converter.